geps/gep-2649/index.md

```diff
@@ -0,0 +1,744 @@
+overridable inside a HTTPRoute), and Overrides cannot be overridden by “lower”
+objects (like setting a maximum client timeout to some non-infinite value at the
+Gateway level to stop HTTPRoute owners from leaking connections over time).
+
+Note that while both types of setting are valid, both also come with difficulties
+in communicating the currently active setting to the owners of the modified
+objects. These are discussed in more detail below.
+
+## Inherited Policy design guidelines
+
+Here are some guidelines for when to consider using a Inherited Policy object:
+
+* The settings or configuration are bound to one containing object, but affect
+  other objects attached to that one (for example, affecting HTTPRoutes attached
+  to a single Gateway, or all HTTPRoutes in a GatewayClass).
+* The settings need to able to be defaulted, but can be overridden on a per-object
+  basis.
+* The settings must be enforced by one persona, and not modifiable or removable
+  by a lesser-privileged persona. (For example, The owner of a GatewayClass may
+  want to restrict something about all Gateways in a GatewayClass, regardless of
+  who owns the Gateway, or a Gateway owner may want to enforce some setting
+  across all attached HTTPRoutes).
+* In terms of status, a good accounting for how to record that the Policy is
+  attached is easy, but recording what resources the Policy is being applied to
+  is not, and needs to be carefully designed to avoid fanout apiserver load.
+
+When multiple Inherited Policies are used, they can interact in various ways,
+which are governed by the following rules, which will be expanded on later in
+this document.
+
+* If a Policy does not affect an object's fields directly, then the resultant
+  Policy MUST be the set of all distinct fields inside the relevant Policy objects,
+  as set out by the rules below.
+* For Policies that affect an object's existing fields, multiple instances of the
+  same Policy Kind affecting an object's fields MUST be evaluated as
+  though only a single Policy "wins" the right to affect each field. This operation
+  is performed on a _per-distinct-field_ basis.
+* Settings in `overrides` stanzas MUST win over the same setting in a `defaults`
+  stanza.
+* `overrides` settings operate in a "less specific beats more specific" fashion -
+  Policies attached _higher_ up the hierarchy beat the same type of Policy
+  attached further down the hierarchy.
+* `defaults` settings operate in a "more specific beats less specific" fashion -
+  Policies attached _lower down_ the hierarchy beat the same type of Policy
+  attached further _up_ the hierarchy.
+* For `defaults`, the _most specific_ value is the one _inside the object_ that
+  the Policy applies to; that is, if a Policy specifies a `default`, and an object
+  specifies a value, the _object's_ value wins.
+* Policies interact with the fields they are controlling in a "replace value"
+  fashion.
```

Comment:
Not sure I'm following this - is this referring to an interaction between a policy and a built-in resource like a Route?

Reply:
Yes. I'll see if I can update to make this clearer, but this is talking about the case where we have a _field_ in an object (like a Route), that is defaulted or overridden by a Policy.